src/plugins/saved_objects_management/public/services/types/column.ts

```diff
@@ -7,9 +7,34 @@
  */
 
 import { EuiTableFieldDataColumnType } from '@elastic/eui';
+import type { Capabilities } from 'src/core/public';
 import { SavedObjectsManagementRecord } from '.';
 
-export interface SavedObjectsManagementColumn<T> {
-  id: string;
-  euiColumn: Omit<EuiTableFieldDataColumnType<SavedObjectsManagementRecord>, 'sortable'>;
+interface ColumnContext {
+  capabilities: Capabilities;
+}
+
+export abstract class SavedObjectsManagementColumn {
```

Comment:
I had to change this from an interface to a class to add the `refreshOnFinish` functionality.
This technically breaks the plugin's contract, but this column service was only added to support Sharing Saved Objects and it has no other consumers.

We could also take this opportunity to remove the column service from the SOM plugin's public contract because it's no longer being consumed by another plugin (all of the logic has been moved to the SOM plugin itself for a while now). However, I didn't want to expand the scope of this PR too much, so I figured I'd leave any optional tech debt reduction to the Core team for another time, since they own this part of the codebase.